libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -476,11 +473,11 @@ std::optional<smtutil::Expression> CHCSmtLib2Interface::invariantsFromSolverResp
 				return first.name < second.name;
 			});
 
-		Expression predicate(asAtom(args[1]), predicateArgs, SortProvider::boolSort);
-		// FIXME: Why do we need to represent the predicate as Expression?
-		definitions.push_back(predicate == parsedInterpretation);
+		std::string const& predicateName = asAtom(args[1]);
+		smtSolverInteractionRequire(!definitions.contains(predicateName), "Predicates must be unique");
+		definitions.insert({predicateName, {parsedInterpretation, predicateArguments}});
```

Comment:
could use `emplace` with `std::piecewise_construct` although not sure how much of a difference it really makes (aside from being harder to read lol)

Reply:
I use `emplace`, but I did not add `piecewise_construct`. Performance is definitely not critical here :)

Reply:
ok fair!